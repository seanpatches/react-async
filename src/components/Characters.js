import React from 'react';
import PropTypes from 'prop-types';
import Character from './Character.js';

function Characters({ characters }) {
  const charactersList = characters.map(character => (
    <li key={character.name}>
      <Character character={character} />
    </li>
  ));


  return ( 
    <ul>
      {charactersList}
    </ul>
  );
}

Characters.propTypes = {
  characters: PropTypes.array.isRequired
};

export default Characters;
