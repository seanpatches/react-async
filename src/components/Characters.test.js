import React from 'react';
import { shallow } from 'enzyme';
import Characters from './Characters';

describe('Characters component', () => {
  it('renders Characters', () => {
    const characters = [{
      name: 'Me',
      status: 'Dead',
      species: 'Dog',
      age: '20'
    },
    {
      name: 'You',
      status: 'Alive',
      species: 'Dog',
      age: '10'
    }];

    const wrapper = shallow(<Characters characters={characters} />);
    expect(wrapper).toMatchSnapshot();
  });
});
