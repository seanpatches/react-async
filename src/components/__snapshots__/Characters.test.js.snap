// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters component renders Characters 1`] = `
<ul>
  <li
    key="Me"
  >
    <Character
      character={
        Object {
          "age": "20",
          "name": "Me",
          "species": "Dog",
          "status": "Dead",
        }
      }
    />
  </li>
  <li
    key="You"
  >
    <Character
      character={
        Object {
          "age": "10",
          "name": "You",
          "species": "Dog",
          "status": "Alive",
        }
      }
    />
  </li>
</ul>
`;
